auto-docs: Update @redpanda-data/docs-extensions-and-macros (#1684)
redpanda-data/documentation: docs/deploy/deployment-option/cloud/vpc-peering-gcp.mdx

---

> title: VPC peering a GCP BYOC cluster

Reviewer: ```suggestion
title: Create a VPC Connection to a GCP BYOC Cluster
```

> title: Create a VPC Connection to a GCP BYOC Cluster

Reviewer 2: ```suggestion
title: Create a VPC Peering Connection to a GCP BYOC Cluster
```

---

<head>

>     <meta name="title" content="VPC Peering with a GCP BYOC Cluster | Redpanda Docs"/>
>     <meta name="description" content="Use the Redpanda and GCP UIs to complete VPC peering on for a BYOC cluster."/>

Reviewer: ```suggestion
    <meta name="title" content="Create a VPC Connection to a GCP BYOC Cluster| Redpanda Docs"/>
    <meta name="description" content="Use the Redpanda and GCP UIs to create a VPC peering connection for a BYOC cluster."/>
```

    <link rel="canonical" href="https://docs.redpanda.com/docs/deploy/deployment-option/cloud/create-byoc-cluster-gcp/" />
</head><br />

> This documentation covers peering your Redpanda network to your existing VPC. This is required to start sending data to Redpanda. 

Reviewer 2: ```suggestion
To start sending data to the Redpanda cluster, you must configure the VPC network connection by connecting your Redpanda VPC to your existing GCP VPC.
```

> ### Prerequisites: 

Reviewer 2: ```suggestion
### Prerequisites
```

* A GCP account 

> * A running BYOC cluster in GCP. If you don't have this ready, follow [this tutorial](create-byoc-cluster-gcp.mdx) first.

Reviewer 2: ```suggestion
* A running BYOC cluster in GCP. See [Create a BYOC Cluster on GCP](create-byoc-cluster-gcp.mdx).
```

> * Make sure that your Redpanda cluster and your VPC are in the same region.

Reviewer 2: ```suggestion
* Your Redpanda cluster and VPC must be in the same region.
```

## Create VPCs 
 
1. Go to the **VPC** section in your GCP project UI. 

> 2. There, you should see an existing VPC. This will have an ID with a `redpanda-` prefix. 

Reviewer 2: ```suggestion
2. You should see an existing VPC. This has an ID with a `redpanda-` prefix. 
```

> 3. If you do not already have a second VPC to connect your Redpanda network to, create one. 
>     
>     * This is your Redpanda client. Ensure that its CIDR does not overlap with the Redpanda network from step 1. 
>     * For the purposes of this tutorial, the name of this will be `rp-client`.  

Reviewer 2: ```suggestion
3. If you don't already have a second VPC to connect your Redpanda network to, create one. 
    
    * This is your Redpanda client. Ensure that its CIDR does not overlap with the Redpanda network from step 1. 
    * The following example uses the name `rp-client`.  
```

## Create a new peering connection 

> 5. In the GCP project UI, go to **Peering Connections**. 
> 6. Create a new peering connection. Fill in the following values:
> 
>     * Your VPC network: `rp-client`
>     * Peered VPC network: `redpanda-[id]`
> 
> 7. Save changes
> 8. Create another peering connection, with the reverse values as above: 
> 
>     * Your VPC network: `redpanda-[id]`
>     * Peered VPC network: `rp-client`
> 
> 9. Save all your changes. GCP should set up routing automatically. 

Reviewer 2: ```suggestion
1. In the GCP project UI, go to **Peering Connections**. 

2. Create a new peering connection with the following values:
    * Your VPC network: `rp-client`
    * Peered VPC network: `redpanda-<id>`
    
3. Save changes.

4. Create another peering connection, with the reverse values as above: 
    * Your VPC network: `redpanda-<id>`
    * Peered VPC network: `rp-client`
    
5. Save changes. GCP should set up routing automatically. 
```

## Connect to Redpanda​

> Your BYOC cluster should now be ready for you to interact with it. On the cluster Overview page, we'll have a variety of ways for you to connect and start sending data. 

Reviewer 2: ```suggestion
The cluster Overview page has a variety of ways for you to connect and start sending data. 
```

> If you want to test this quickly in GCP, you can: 
> 

Reviewer 2: ```suggestion
To quickly test this quickly in GCP: 
```

> * Create a Virtual Machine on your GCP network that has a firewall rule allowing ingress traffic from your IP (e.g. `[your ip here]/32`)

Reviewer: ```suggestion
* Create a Virtual Machine on your GCP network that has a firewall rule allowing ingress traffic from your IP (e.g. `<your ip>/32`)
```

> * Activate the Cloud Shell in your project, install `rpk` in the Cloud Shell, and run `rpk cluster info`
> * If there is output from Redpanda, your connection has been successful 

Reviewer 2: ```suggestion
* Activate the Cloud Shell in your project, install `rpk` in the Cloud Shell, and run `rpk cluster info`.
* If there is output from Redpanda, your connection is successful. 
```
